R/loadFromMothur.R

```diff
@@ -0,0 +1,206 @@
+    } else {
+      sample_meta <- S4Vectors:::make_zero_col_DataFrame(ncol(feature_tab))
+    }
+    
+    if (!is.null(phyTreeFile)) {
+      tree <- .read_qza(phyTreeFile, ...)
+    } else {
+      tree <- NULL
+    }
+    
+    # if row.names(feature_tab) is a DNA sequence,  set it as refseq
+    if (!is.null(refSeqFile)){
+      refseq <- .read_qza(refSeqFile, ...)
+    } else if (featureNamesAsRefSeq) {
+      refseq <- .rownames_as_dna_seq(rownames(feature_tab))
+    } else {
+      refseq <- NULL
+    }
+    
+    return(TreeSummarizedExperiment(
+      assays = S4Vectors::SimpleList(counts = feature_tab),
+      rowData = taxa_tab,
+      colData = sample_meta,
+      rowTree = tree,
+      referenceSeq = refseq
+    ))
+    
+}
+
+.read_mothur_feature <- function(featureTableFile, ...){
+
+    # Reads the file
+    assay <- read.table(featureTableFile, check.names=FALSE, header=TRUE,
+                      sep="\t", stringsAsFactors=FALSE)
+  
+    # File contains additional columns "label", "numOtus", and "Group". They are removed
+    assay$label <- NULL
+    assay$numOtus <- NULL
+    # Information of "Group" column is saved, it includes taxa information
+    x <- assay$Group
+    assay$Group <- NULL
+    
+    # Initializes rownames
+    rownames(assay) <- NULL
+    # Saves taxa to rownames
+    rownames(assay) <- x
+    
+    # Transposes and converts assay from dataframe to matrix
+    assay <- t(as.matrix(assay))
+    
+    return(assay)
+    
+}
+
+.read_mothur_taxonomy <- function(taxonomyTableFile, feature_tab, ...){
+    
+    # Reads the file
+    rowData <- read.table(taxonomyTableFile, check.names=FALSE,
+                          header=TRUE, sep="\t", stringsAsFactors=FALSE)
+    # Deletes additional information between taxa levels.
+    rowData$Taxonomy <- gsub("[\"]", "", rowData$Taxonomy)
+    rowData$Taxonomy <- gsub("[(1-100)]", "", rowData$Taxonomy)
+    # Separate taxa to own columns
+    rowData <- tidyr::separate(rowData, "Taxonomy", into=c("Kingdom",
+                                                           "Phylum", "Order", "Class", "Family", "Genus"), sep=";",
+                               extra="merge")
+    # Deletes ";" from the end of Genus names
+    rowData$Genus <- gsub(";", "", rowData$Genus)
+    # Deletes addition "Size" column
+    rowData$Size <- NULL
+    
+    # Adds taxa to rownames
+    rownames(rowData) <- rowData$OTU
+    # Deletes addition "OTU" column
+    rowData$OTU <- NULL
+    
+    # Creates a matrix from the table
+    rowData <- as.matrix(rowData)
+    
+    # Gets only those taxa that are included in feature_tab
+    rowData <- rowData[rownames(feature_tab),]
+    
+    return(rowData)
+
+}
+
+.read_mothur_sample_meta <- function(sampleMetaFile, feature_tab, ...){
+  
+    # Reads the file
+    colData <- read.csv(sampleMetaFile, row.names=1, check.names=FALSE)
+    # Gets only those samples that are included in feature_tab
+    colData <- colData[colnames(feature_tab), ]
```

Comment:
Same as above. Why don't the dimension match? Or the order?

I don't think that this hand-holding is necessary. If mothur supplies data which matches, we don't need to make sure, the user didn't jumble the data around.

If mothur supplies jumbled-up data, then I see the point.